buildscripts/kokoro/xds-k8s-install-test-driver.sh

```diff
@@ -0,0 +1,362 @@
+# Export Keystore secrets assigned to the Kokoro build.
+# Globals:
+#   KOKORO_KEYSTORE_DIR
+#   PRIVATE_API_KEY: Exported. Populated with name GCP project secret key.
+#                    Used by the test driver to access private APIs
+# Arguments:
+#   None
+#######################################
+kokoro_export_secrets() {
+  readonly PRIVATE_API_KEY=$(cat "${KOKORO_KEYSTORE_DIR}/73836_grpc_xds_interop_tests_gcp_alpha_apis_key")
+  export PRIVATE_API_KEY
+}
+
+#######################################
+# Configure Python virtual environment on Kokoro VM.
+# Arguments:
+#   None
+# Outputs:
+#   Writes the output of `pyenv` commands to stdout
+#######################################
+kokoro_setup_python_virtual_environment() {
+  # Kokoro provides pyenv, so use it instead `python -m venv`
+  echo "Setup pyenv environment"
+  eval "$(pyenv init -)"
+  eval "$(pyenv virtualenv-init -)"
+  py_latest_patch="$(pyenv versions --bare --skip-aliases | grep -E "^${PYTHON_VERSION}\.[0-9]{1,2}$" | sort --version-sort | tail -n 1)"
+  echo "Activating python ${py_latest_patch} virtual environment"
+  pyenv virtualenv "${py_latest_patch}" k8s_xds_test_runner
+  pyenv local k8s_xds_test_runner
+  pyenv activate k8s_xds_test_runner
+}
+
+#######################################
+# Installs and configures the test driver on Kokoro VM.
+# Globals:
+#   KOKORO_ARTIFACTS_DIR
+#   TEST_DRIVER_REPO_NAME
+#   SRC_DIR: Populated with absolute path to the source repo on Kokoro VM
+#   TEST_DRIVER_REPO_DIR: Populated with the path to the repo containing
+#                         the test driver
+#   TEST_DRIVER_FULL_DIR: Populated with the path to the test driver source code
+#   TEST_DRIVER_FLAGFILE: Populated with relative path to test driver flagfile
+#   TEST_XML_OUTPUT_DIR: Populated with the path to test xUnit XML report
+#   KUBE_CONTEXT: Populated with name of kubectl context with GKE cluster access
+#   PRIVATE_API_KEY: Populated with name GCP project secret key. Used by the
+#                    test driver to access private APIs
+#   GIT_ORIGIN_URL: Populated with the origin URL of git repo used for the build
+#   GIT_COMMIT_SHORT: Populated with the short SHA-1 of git commit being built
+# Arguments:
+#   The name of github repository being built
+# Outputs:
+#   Writes the output to stdout, stderr, files
+#######################################
+kokoro_setup_test_driver() {
+  local src_repository_name="${1:?Usage kokoro_setup_test_driver GITHUB_REPOSITORY_NAME}"
+  # Capture Kokoro VM version info in the log.
+  kokoro_print_version
+
+  # Kokoro clones repo to ${KOKORO_ARTIFACTS_DIR}/github/${GITHUB_REPOSITORY}
+  local github_root="${KOKORO_ARTIFACTS_DIR}/github"
+  readonly SRC_DIR="${github_root}/${src_repository_name}"
+  local test_driver_repo_dir="${github_root}/${TEST_DRIVER_REPO_NAME}"
+  parse_src_repo_git_info SRC_DIR
+  kokoro_write_sponge_properties
+
+  # Turn off command trace print before exporting secrets.
```

Comment:
Having to do this at all looks pretty scary. Instead of exporting the secret key contents, can we just put the path to the key file in PRIVATE_API_KEY and load it [here](https://github.com/grpc/grpc/blob/e2f69ca315de53f627990874705a53e79b317793/tools/run_tests/xds_k8s_test_driver/framework/infrastructure/gcp/api.py#L59)?

Reply:
It is possible to do that. However, I see the following tradeoffs:
1) Passing file path to with the file key content is creating a tight couple with this specific implementation. On the other hand, using environment variables for runtime configuration is [quite common](https://12factor.net/config)
2) Additional security benefits from using file are not exactly clear. If an attacker can execute a code to read an env war, what would stop them from reading the file.

Also, this secret essentially can't be used for anything else than reading discovery documents of two private APIs (which are available in gcloud source anyway).

Reply:
I'm not concerned about an attacker executing code to read an env var; I'm concerned about us accidentally leaking the contents of the private key into publicly accessible log output. What the secret is currently used for is also not important here, as its scope could be expanded in the future. Requiring anyone working with this environment variable to have to remember, as done here, to turn off command tracing seems very error-prone.

I don't think concerns about tight coupling with this implementation (I'm not sure I understand what that means here, anyways - we have cross-repo shell scripts that are not exactly independent already) outweigh the potential risk of dumping the raw text of the api key into public logs.

Reply:
To clarify: I meant tight coupling introduced in the driver itself - reading contents of a file with to get a secret seems a bit arbitrary. Ideally the driver shouldn't care if it's ran on Kokoro or anywhere else, and having secret from unstructured file seems a bit too specific to how Kokoro handles keystore.

Nevertheless, I absolutely agree with accidentally leaking secrets in public logs is much worse. Plus, this would set a bad precedent.

I'll give this a thought, and come back here.

Reply:
As discussed in gchat, replaced the env var with a secret from Cloud Secret Manager: grpc/grpc#25042

Reply:
Thanks. The entire turn off/turn on command trace (+x/-x) logic here can now be removed? Or at least the comments updated. Or push into  kokoro_setup_python_virtual_environment if noise from pyenv is also an issue?